plugin/utils/gapic_utils.py

```diff
@@ -37,17 +37,38 @@ def read_from_gapic_yaml(yaml_file):
         gapic_yaml = yaml.load(f)
 
     collections = {}
+    fixed_collections = {}
+
+    all_collections = []
     if 'collections' in gapic_yaml:
-        collections = load_collection_configs(gapic_yaml['collections'],
-                                              collections)
+        all_collections = [gapic_yaml['collections']]
     for interface in gapic_yaml.get('interfaces', []):
         if 'collections' in interface:
-            collections = load_collection_configs(interface['collections'],
-                                                  collections)
-    fixed_collections = {}
+            all_collections.append(interface['collections'])
+
+    all_entities = []
+    for collection in all_collections:
+        all_entities.extend(collection)
+
+    (single_resource_names, fixed_resource_names) = \
+        find_single_and_fixed_entities(all_entities)
+
+    collections = load_collection_configs(single_resource_names, collections)
+
+    # TODO(andrealin): Remove the fixed_resource_name_values
+    # parsing once they no longer exist in GAPIC configs.
     if 'fixed_resource_name_values' in gapic_yaml:
         fixed_collections = load_fixed_configs(
-            gapic_yaml['fixed_resource_name_values'], collections)
+            fixed_collections,
```

Comment:
Are first two args reversed?

Reply:
whoops
added another test to catch this, and also to test the parsing the status quo `fixed_resource_name_values` from the GAPIC config.